src/solver/variable/economy/max-mrg.h

```diff
@@ -246,7 +240,11 @@ class Marge : public Variable::IVariable<Marge<NextT>, NextT, VCardMARGE>
     void weekForEachArea(State& state, unsigned int numSpace)
     {
         double* rawhourly = Memory::RawPointer(pValuesForTheCurrentYear[numSpace].hour);
-        PrepareMaxMRG(state, rawhourly + state.hourInTheYear, numSpace);
+
+        // Getting data required to compute max margin
+        MaxMrgUsualDataFactory maxMRGdataFactory(state, numSpace);
+        MaxMRGinput maxMRGinput = maxMRGdataFactory.data();
+        computeMaxMRG(rawhourly + state.hourInTheYear, maxMRGinput);
```

Comment:
Note that function **PrepareMaxMRG** was renamed into **computeMaxMRG**.
It takes less arguments (arg **numspace** is gone), and arguments roles are clearer (data in and result out).

Instead of one call to **PrepareMaxMRG**, we have more step.
- creation of a factory to build input data for **MAX MRG** computation algorithm
- Building the input data for **MAX MRG** computation
- Compute the **MAX MRG**